### PR TITLE
feat: add Hygraph support

### PR DESCRIPTION
**Pull request by mattwigham**
Thanks for the great lib.

### CODE REVIEW

**Comment by ascorbic** [comment]
Thanks! Looking at the playground it doesn't seem to be cropping the images as expected. e.g. making this 100x600 gives a 100x100 image. Is there a param that can give the expected aspect ratio? The default should be equivalent to object-fit: cover, if it's supported.

**Comment by ascorbic** [comment]
Thanks. Can you enable edits from maintainers on the PR please

**Comment by mattwigham** [comment]
Hmm, apparently I don't have that option because the fork is on an organization. https://github.com/orgs/community/discussions/5634

**Comment by ascorbic** [comment]
That's annoying. Can you update the branch then please. There should be a button just below this

**Comment by mattwigham** [comment]
Done!

**Review by ascorbic — APPROVED** [review]
Thanks!

File: src/transformers/hygraph.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+) => {
+	const parsed = parse(base.toString());
+	const props: HygraphParams = {
+		transformations: {},
+		...parsed.params,
+		...params,
+	};
+
+  props.transformations.resize ||= {};
+  props.transformations.resize.fit ||= "crop";
```
Comment (ascorbic) [inline src/transformers/hygraph.ts:53]: This needs to only be set if both width and height are provided. I think your previous version had a check for this. Currently this generates invalid URLs if only one dimension is provided. e.g. [this one](https://us-west-2.graphassets.com/cm2apl1zp07l506n66dmd9xo8/resize=fit:crop,width:100/auto_image/cm2tr64fx7gvu07n85chjmuno). You can catch errors like this by running `deno test --allow-net e2e.test.ts`

**Review by ascorbic — CHANGES_REQUESTED** [review]

**Review by ascorbic — APPROVED** [review]

**Pull request merged**